lib/ansible/modules/replace.py

```diff
@@ -284,7 +284,10 @@ def main():
         section = contents
 
     mre = re.compile(params['regexp'], re.MULTILINE)
-    result = re.subn(mre, params['replace'], section, 0)
+    try:
+        result = re.subn(mre, params['replace'], section, 0)
+    except re.error as s:
+        module.fail_json(msg=str(s))
```

Comment:
Maybe something like `module.fail_json(msg="replace failed: %s" % s, exception=format_exc())` - that will preserve the exception detail as well. Technically, the `compile` could also fail in the same way, so should maybe have a similar exception handler there.